### ISSUE TITLE
Interaction metas can be hidden in postponed typechecking problems (Hole position off )

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agda/agda#1083]
``` agda
data Bool : Set where true false : Bool

T : Bool → Set
T true = Bool
T false = Bool → Bool

postulate
  f : {x : Bool} → T x

test : (x : Bool) → T x
test true = f {!!}
test false = ?
```

Since f is expecting a hidden and not an explicit argument, there is yellow 
(for the unresolved hidden meta).  Further, since f's type is blocked by that 
meta, it could be a function type, so applying f to a hole {!!} is not 
certainly wrong.

However, this hole seems not being communicated to emacs, which only shows one 
hole, with the type of the second hole:

```
?0 : T false
_x_9 : Bool  [ at /home/abel/tmp/Agda2/test/bugs/HoleNotGoal.agda:13,13-14 ]
_11 : Bool  [ at /home/abel/tmp/Agda2/test/bugs/HoleNotGoal.agda:13,13-19 ]
```

Original issue reported on code.google.com by `andreas....@gmail.com` on 14 Mar 2014 at 5:31
